lib/reservation.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'date_range'
+
+module Hotel
+	class Reservation
+		attr_reader :id, :room, :block, :rate, :date_range
+
+		def initialize(id:, room:, block: false, rate: 200.0, start_date:, end_date:)
+			@id = id
+			@room = room
+			@block = block
+			@rate = rate.to_f
+			@date_range = Hotel::DateRange.new(
+				start_date: start_date, 
+				end_date: end_date
+			)
+
+			raise ArgumentError.new("Invalid rate!") if rate.class != Float
```

Comment:
If you're going to check types you should probably use `kind_of?` to allow subclassing.

```suggestion
			raise ArgumentError.new("Invalid rate!") unless rate.kind_of?(Float)
```